### PR TITLE
feat(devtools): Setup commitlint

### PR DESCRIPTION
**Pull request by bastilian**
This sets up `commitlint`[0] and proposes rules to lint commit messages.

It adds `npm run lint:commit` as a command to validate commits made in a branch compared `master`.

[0] https://commitlint.js.org/

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by skateman** [comment]
@bastilian can you elaborate on what rules this linter enforces?

**Comment by bastilian** [comment]
@skateman Of course. The (default) rules are following the conventional commit[0] format, that we already use.

 Since we do have some deviation already there are two customised rules. 
The `body-max-line-length`, because dependabots commits have longer lines in the body.
And `subject-case`, because we tend to use component names as scope/subject and the default expects snake-case.

[0] https://www.conventionalcommits.org/en/v1.0.0/

File: commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  extends: ['@commitlint/config-conventional'],
+  rules: {
+    'body-max-line-length': [2, 'always', 150],
```
Comment (vkrizan) [inline commitlint.config.js:4]: How we came up with 150 chars?

How about:
```
[1, 'always', 80]
```
(warning after 80 chars)

Commits can contain URLs that are not breakable.

Comment (bastilian) [reply commitlint.config.js:4]: This is only to satisfy commits by dependabot, we can of course switch to warning after 80.

**Review by marleystipich2 — COMMENTED** [review]
I've tested this and commitlint is working great for me!
Quick question though @bastilian, I'm not seeing an error or warning message when I use a subject line that is all caps, which I think should go against the rules, like `git commit -m "fix: TEST"`. Is this intentional?
I am seeing the correct warning message when I test a body line length over 80 though.

**Comment by bastilian** [comment]
@marleystipich2 Thank you for testing! Yes, this is intentionally disabled via the `subject-case` setting and allows any case, instead of the default, which is only lower-case. The reason for disabling it is to allow us to use the Jira card ID, which is upper-case and not trigger the rule. There doesn't seem to be any other way to allow these.

**Review by marleystipich2 — APPROVED** [review]
Oh good idea! Thanks for the explanation, @bastilian. This works great for me.

**Comment by skateman** [comment]
@vkrizan are you okay with this?

**Comment by bastilian** [comment]
> @vkrizan are you okay with this?

We can always adjust the settings. The current settings are to let through what we have used do far. I took the last 100 commits or so to validate and make sure the pass.

**Pull request merged**